flow/project.py

```diff
@@ -2036,6 +2019,30 @@ def _get_cached_status(self):
         except KeyError:
             return {}
 
+    @contextlib.contextmanager
+    def _update_cached_scheduler_status(self):
```

Comment:
I can't remember, what did we do before about updating the project's status?

Reply:
It was done in a somewhat ad-hoc manner by several methods that modified individual keys in `project.document["_status"]`. The previous code could trigger multiple write operations to the project document, when we really only need one write operation at the end.